vignettes/tractable-autocorrelations.Rmd

```diff
@@ -0,0 +1,108 @@
+sarica$subjectID <- unclass(factor(sarica$subjectID))
+```
+
+We will first fit a GAM model without any autocorrelation structure using the 
+`tractable_single_bundle` function. This model will use "group" and "age" to 
+predict FA, while smoothing over the tract nodes. We will also use the automated
+procedure implemented in  `tractable_single_bundle` to determine the ideal value
+for `k`, a parameter used to determine the number of spline functions. The default behavior for `tractable_single_bundle` is to include the AR1 model. To avoid this, we set the parameter `autocor` to `FALSE`. 
+
+
+```{r fit no ac model}
+gam_fit_cst_no_ac <- tractable::tractable_single_bundle(
+  df_afq         = sarica,
+  tract          = "Right Corticospinal",
+  participant_id = "subjectID",
+  group_by       = "group",
+  covariates     = c("age", "group"),
+  dwi_metric     = "fa",
+  k              = "auto",
+  autocor        = FALSE
+)
+```
+
+Examining the summary of the resulting GAM fit object shows us that the `k=16`
+is sufficiently large to describe the spatial variation of tract profile data.
+In addition, we see that there is a statistically significant effect of group
+(with a p-value of 4.66e-10) and no statistically significant effect of age
+(p=0.2748).
+
+```{r get no ac summary}
+summary(gam_fit_cst_no_ac)
+```
+
+To account for potential spatial autocorrelation of FA values along the length 
+of the tract profile, we can incorporate a AR1 model into our GAM. Briefly, this
+AR1 model is a linear model that estimates the amount of influence of the FA value of the preceding node on the FA value of the current node (see [@VanRij2019](https://journals.sagepub.com/doi/pdf/10.1177/2331216519832483) for an overview of accounting for autocorrelation using the `mgcv` package). 
+
+The AR1 model takes a parameter $\rho$ to estimate autocorrelation effects. We 
+can pass our initial model into the function `itsadug::start_value_rho` to 
+automatically determine the value of $\rho$. We can also plot the autocorrelation
+by setting `plot=TRUE` within that function (pictured below). 
+
+```{r calculate rho}
+rho = itsadug::start_value_rho(gam_fit_cst_no_ac)
+itsadug::start_value_rho(gam_fit_cst_no_ac, plot=T)
+```
+
+By default, the `tractable_single_bundle` function will determine the value of 
+$\rho$ and incorporate the AR1 model into the GAM estimation. 
+
+```{r fit model including AR structure}
+gam_fit_cst <- tractable::tractable_single_bundle(
+  df_afq         = sarica,
+  tract          = "Right Corticospinal",
+  participant_id = "subjectID",
+  group_by       = "group",
+  covariates     = c("age", "group"),
+  dwi_metric     = "fa",
+  k              = "auto"
+)
+```
+
+Examining the summary of the resulting GAM fit object shows us that the inclusion
+of the AR1 model changes the resulting statistics of our model. Although there 
+is still a statistically significant effect of group (p=0.044), the value of the
```

Comment:
That's quite a change in p-value! Interesting 🤔

Reply:
Yeah it's interesting how dramatically things can change based on rho...If it would be useful, I have some code that I can add to this vignette that plots how various statistics (fREML, p-values, k-values, etc...) change based on the value of rho

Reply:
I think it would be interesting to maybe show the ACF for this data - it helps understand why the effect is so dramatic. Maybe even use itsadug directly to show that.